### ISSUE TITLE
3D Secure authentication returns error on user cancellation

### ISSUE THREAD

**Issue by Drusy** [stripe/stripe-ios#1352]
### Unexpected behavior

3D Secure authentication always fails even when user action is cancellation.
The callback of `STPPaymentHandler.shared().confirmPayment` returns a `STPPaymentHandlerActionStatus` which should be `.canceled` in case of user cancellation. Since v17.0, cancelling a 3D Secure action returns `.failed` with the following error:

```
 Error Domain=STPPaymentHandlerErrorDomain Code=5 "We are unable to authenticate your payment method. Please choose a different payment method and try again." UserInfo={transaction_status=N, NSLocalizedDescription=We are unable to authenticate your payment method. Please choose a different payment method and try again.}
```

### Expected behavior

In case of cancellation, `confirmPayment` should return  `STPPaymentHandlerActionStatus.canceled`

### Test card used

4000 0000 0000 3220

Or any card showing 3D Secure in a native view. Webview based authentication `close` button seems to return a `.canceled` callback

## iOS version
iOS 12.4 | iOS 12.3.1

## Installation method
Carthage 

## SDK version
> github "stripe/stripe-ios" "v17.0.0"

**Comment by Drusy** [comment]
#1268 and #1271 seems to be related, is this the expected behavior ?

**Comment by yuki-stripe** [comment]
Hi @Drusy,

Thanks for the report, I've reproduced this on my end.  This is a regression and we're working on a fix now.

**Comment by treatwell-ignasurbonas** [comment]
This one is labeled as `fixed` but it still happens on stripe SDK`v17.0.0.`.

**Comment by Drusy** [comment]
@treatwell-ignasurbonas The issue has been reported on v17.0.0 so this is normal. 
The issue will be closed on the next release I guess, but has been fixed in the codebase by the Stripe team merging this PR: https://github.com/stripe/stripe-ios/pull/1353

**Comment by davidme-stripe** [comment]
This has been fixed in Stripe SDK 17.0.1.
